thirtyinch/src/main/java/net/grandcentrix/thirtyinch/internal/TiFragmentDelegate.java

```diff
@@ -0,0 +1,256 @@
+            @Nullable final Bundle savedInstanceState) {
+        mViewBinder.invalidateView();
+    }
+
+    public void onDestroy() {
+        //FIXME handle attach/detach state
+
+        logState();
+
+        boolean destroyPresenter = false;
+        if (mTiFragment.isActivityFinishing()) {
+            // Probably a backpress and not a configuration change
+            // Activity will not be recreated and finally destroyed, also destroyed the presenter
+            destroyPresenter = true;
+            TiLog.v(mLogTag.getLoggingTag(),
+                    "Activity is finishing, destroying presenter " + mPresenter);
+        }
+
+        final TiConfiguration config = mPresenter.getConfig();
+        if (!destroyPresenter &&
+                !config.shouldRetainPresenter()) {
+            // configuration says the presenter should not be retained, a new presenter instance
+            // will be created and the current presenter should be destroyed
+            destroyPresenter = true;
+            TiLog.v(mLogTag.getLoggingTag(),
+                    "presenter configured as not retaining, destroying " + mPresenter);
+        }
+
+        if (!destroyPresenter &&
+                !config.useStaticSaviorToRetain() && mTiFragment.isDontKeepActivitiesEnabled()) {
+            // configuration says the PresenterSavior should not be used. Retaining the presenter
+            // relays on the Activity nonConfigurationInstance which is always null when
+            // "don't keep activities" is enabled.
+            // a new presenter instance will be created and the current presenter should be destroyed
+            destroyPresenter = true;
+            TiLog.v(mLogTag.getLoggingTag(),
+                    "the PresenterSavior is disabled and \"don\'t keep activities\" is "
+                            + "activated. The presenter can't be retained. Destroying "
+                            + mPresenter);
+        }
+
+        if (destroyPresenter) {
+            mPresenter.destroy();
+            PresenterSavior.INSTANCE.free(mPresenterId);
+        } else {
+            TiLog.v(mLogTag.getLoggingTag(), "not destroying " + mPresenter
+                    + " which will be reused by the next Activity instance, recreating...");
+        }
+    }
+
+    public void onDestroyView() {
+        mPresenter.detachView();
+    }
+
+    public void onSaveInstanceState(final Bundle outState) {
+        outState.putString(SAVED_STATE_PRESENTER_ID, mPresenterId);
+    }
+
+    public void onStart() {
+        mActivityStarted = true;
+
+        if (isUiPossible()) {
+            mTiFragment.postToMessageQueue(new Runnable() {
+                @Override
+                public void run() {
+                    if (isUiPossible() && mActivityStarted) {
+                        mViewBinder.bindView(mPresenter, mViewProvider);
+                    }
+                }
+            });
+        }
+    }
+
+    public void onStop() {
```

Comment:
In our `TiActivityDelegate` we have called these method with the suffix `_afterSuper`. 
It would be great if you either
* Add this prefix here too.
* Remove the prefix from the `TiActivityDelegate`
It is more consistent...